debezium-reactive/src/main/java/io/debezium/reactive/DebeziumEvent.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.reactive;
+
+import org.apache.kafka.connect.source.SourceRecord;
+
+/**
+ * A class that encapsulates {@link SourceRecord} and associates the {@link #commit()} operation with it.
+ * It is expected that the consumer will receive the event from the pipeline and when the event is
+ * durably processed it will signal this fact back to engine using {@link #commit()} call.
+ *
+ * @author Jiri Pechanec
+ *
+ */
+public interface DebeziumEvent {
+
+    /**
+     * @return Change data capture event as {@link SourceRecord} received from Debezium connector.
+     */
+    public SourceRecord getRecord();
```

Comment:
It'd be nice to not tie this API to `SourceRecord` unconditionally, so people can consume this as JSON or even Avro.

It'd be cool if we could parameterize the type (`DebeziumEvent<T>`), and when bootstrapping the engine, you could give a marker class which links the expected event record type and the corresponding serializer implementation which needs to be called.

E.g. like this:

```java
ReactiveEngine<String> engine = create()
    .withConfiguration(config)
    .eventType(Json.class) // this would return Builder<String>
    .build();
```

And

```
public class Json extends ReactiveEngineEventType<String> {
    Class<String> getEventType() {
        return String.class;
    }

    Class<JsonSerializer> getSerializerType() {
        return JsonSerializer.class; // JsonSerializer extends Serializer<String> { ... }
    }
}
```